cip-TOKEN-STD/cip-meiersi-Canton-Network-Token-Standard.md

```diff
@@ -0,0 +1,567 @@
+OpenAPI.
+
+The following diagram provides an overview of the interactions enabled by these
+APIs.
+
+![API interaction diagram](images/api-diagram.png)
+
+
+#### Implementation Requirements
+
+The code in the [Reference implementation](#reference-implementation) provides
+the exact definitions of the APIs together with inline comments specifying the
+exact contracts between API clients and implementors.
+
+Asset registries can in principle freely choose which of the five APIs
+concerning them they want to implement. However, they should implement all five
+of them to maximize the utility of their tokens. The exception are asset
+registries whose authoritative holding records are not maintained on Canton.
+They may decide to not implement the "holdings API" to avoid stale holding
+records on Canton.
+
+Wallets and custody solutions should aim to make use of all six APIs to provide
+a uniform UX across all Canton Network tokens. They should query a registry's
+metadata to figure out which APIs it supports and adjust their UI accordingly.
+
+Apps may use the "allocation API" to orchestrate asset transfers as part of
+their own workflows. They may also implement the "allocation request API"
+to integrate with wallets in a uniform way.
+
+##### Global Synchronizer Connectivity for Settlement Workflows
+
+This CIP recommends registries to maximize the utility of their assets by
+connecting to the Global Synchronizer (GS) and supporting the execution of
+settlement workflows with their assets on the GS.  This is only a
+recommendation. They are free to use the token standard APIs on private
+synchronizers as they see fit.
+
+The reason for this recommendation is that the input contracts (i.e., UTXOs) referenced by a Daml transaction
+must all be assigned to the same synchronizer.
+Settlement transactions likely involve multiple registries, holders, and apps.
+Therefore the most likely synchronizer that all input contracts of a settlement transaction
+can be assigned to is the GS.
+Thus the benefit of connecting to the GS for settlement workflows is that this
+maximizes the chance of there being a synchronizer that can synchronize
+the Daml transactions required for the settlement to complete.
+
+
+#### Off-Ledger API Discovery and Access
+
+The standard expects registry apps to expose the standard's HTTP endpoints for
+accessing UTXOs to the public internet under a common URL prefix to provide
+maximal freedom for wallets and apps to fetch these.
+
+The standard does not require requests to these HTTP endpoints be authenticated, as all of them
+either access data that is expected to be public (e.g., registry metadata) or
+data that is protected by a difficult to guess identifier (e.g., the contract-id
+of an allocation). See [this section](#no-authentication-on-registry-off-ledger-apis)
+for the rationale.
+
+Decentralized registries (like Amulet) must serve these HTTP endpoints from each of their nodes
```

Comment:
do we really want to make this a "must"? My understanding from the iBTC discussions was the opposite. If they do want to have DA provide the off ledger endpoint we're not really opposed. We've done the work through `expectedAdmin` that this is relatively sane from a security pov.